utils/S3UtilsMongoClient.js

```diff
@@ -14,51 +14,57 @@ const __COUNT_ITEMS = 'countitems';
-                'value.isNull': 1,
-                'value.archive': 1,
-                'value.x-amz-storage-class': 1,
-                'value.isPHD': 1,
-            },
-        });
-        const collRes = {
-            bucket: {}, // bucket level metrics
-            location: {}, // location level metrics
-            account: {}, // account level metrics
-        };
-        let stalledCount = 0;
-        const cmpDate = new Date();
-        cmpDate.setHours(cmpDate.getHours() - 1);
-
-        const locationConfig = getLocationConfig(log);
+    async getObjectMDStats(bucketName, bucketInfo, isTransient, log, callback) {
+        try {
+            const c = this.getCollection(bucketName);
+            const cursor = c.find({}, {
+                projection: {
+                    '_id': 1,
+                    'value.last-modified': 1,
+                    'value.replicationInfo': 1,
+                    'value.dataStoreName': 1,
+                    'value.content-length': 1,
+                    'value.versionId': 1,
+                    'value.owner-id': 1,
+                    'value.isDeleteMarker': 1,
+                    'value.isNull': 1,
+                    'value.archive': 1,
+                    'value.x-amz-storage-class': 1,
+                    'value.isPHD': 1,
+                },
+            });
+            const collRes = {
+                bucket: {}, // bucket level metrics
+                location: {}, // location level metrics
+                account: {}, // account level metrics
+            };
+            let stalledCount = 0;
+            const cmpDate = new Date();
+            cmpDate.setHours(cmpDate.getHours() - 1);
+
+            const locationConfig = getLocationConfig(log);
+
+            const usersBucketCreationDatesArray = await this.getCollection(USERSBUCKET).find({}, {
+                projection: {
+                    'value.creationDate': 1,
+                },
+            }).toArray();
 
-        return this.getCollection(USERSBUCKET).find({}, {
-            projection: {
-                'value.creationDate': 1,
-            },
-        }).toArray((err, usersBucketCreationDatesArray) => {
-            if (err) {
-                log.error('Failed to get users bucket creation dates', {
-                    method: 'getObjectMDStats',
-                    error: err,
-                });
-                return callback(err);
-            }
             const usersBucketCreationDatesMap = usersBucketCreationDatesArray
                 .reduce((map, obj) => ({ ...map, [obj._id]: obj.value.creationDate }), {});
-            return cursor.forEach(
+            const startCursorDate = new Date();
+            let processed = 0;
+            await cursor.forEach(
                 res => {
+                    // Periodically display information about the cursor
+                    // if more than 30s elapsed
+                    if (Date.now() - startCursorDate > 30000) {
```

Comment:
this will call each time the callback is called, once the 30 second liimit has been reached: need to reset `startCursorDate` when we add a log ?

```suggestion
                    const currentDate = Date.now();
                    if (currentDate - startCursorDate > 30000) {
                        startCursorDate = currentDate;
```

Reply:
Done in https://github.com/scality/s3utils/commit/8f7df21ab49415c7a5bb523ac6ffb5a1beaa2084

Reply:
I rebased and squashed the commits, see the updated link: https://github.com/scality/s3utils/pull/300/files#diff-3a778b361ddfba6106347ff0cb053bcbb1a84374fa0d7e6e6413e7c6fb6ca5cdR55